lib/cli-engine.js

```diff
@@ -174,51 +172,26 @@ function processText(text, configHelper, filename, fix, allowInlineConfig, linte
         }
     }
 
-    if (processor) {
-        debug("Using processor");
-        const parsedBlocks = processor.preprocess(text, filename);
-        const unprocessedMessages = [];
-
-        parsedBlocks.forEach(block => {
-            unprocessedMessages.push(linter.verify(block, config, {
-                filename,
-                allowInlineConfig
-            }));
-        });
-
-        // TODO(nzakas): Figure out how fixes might work for processors
-
-        messages = processor.postprocess(unprocessedMessages, filename);
-
-    } else {
-
-        if (fix) {
-            fixedResult = linter.verifyAndFix(text, config, {
-                filename,
-                allowInlineConfig,
-                fix
-            });
-            messages = fixedResult.messages;
-        } else {
-            messages = linter.verify(text, config, {
-                filename,
-                allowInlineConfig
-            });
-        }
-    }
+    const fixedResult = linter.verifyAndFix(text, config, {
+        filename,
+        allowInlineConfig,
+        fix: typeof fix !== "undefined" && fix,
```

Comment:
This would result in performance impact, I would guess. Now we are trying to fix every error, even if we don't need to.

Reply:
I don't think so -- `SourceCodeFixer` has a fast path to just skip all fixes when `fix: false` is used.

Reply:
I mean in the case of processor not supporting autofixing (and not wanting to support it in the future). Autofixing will still happen, but it's results will be ignored by `postprocess` function.

Reply:
Oh, I see. This could be addressed by making autofixing opt-in for processors (https://github.com/eslint/eslint/pull/9090#issuecomment-326775445). Currently, fixes would still get applied by ESLint in this case, but the processor would usually generate broken code.

Reply:
Opt-in sounds fine to me.